metrics-api/src/main/java/software/amazon/swage/metrics/MetricContext.java

```diff
@@ -13,7 +13,6 @@
  * <p/>
  * A {MetricContext} is the primary interface used to record measurements.
  */
-// TODO: provide a childContext() mechanism?
 public interface MetricContext extends AutoCloseable {
```

Comment:
I probably missed the boat on this one, but:
I'm not sure calling this a 'MetricContext' is strictly the best idea.  It is conceptually a context in the larger sense, that only happens to be very useful for the metrics stuff here - one of the design intents was for it to be a more widely usable context (for things such as request tracking etc.) that just happened to be needed/used for metrics first.